drivers/charger/bq24190.c

```diff
@@ -0,0 +1,485 @@
+/*
+ * Copyright 2023 Cirrus Logic, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_bq24190
+
+#include <errno.h>
+
+#include "bq24190.h"
+
+#include "zephyr/device.h"
+#include "zephyr/drivers/charger.h"
+#include "zephyr/drivers/i2c.h"
+#include "zephyr/kernel.h"
+#include "zephyr/sys/util.h"
+#include "zephyr/logging/log.h"
+
+LOG_MODULE_REGISTER(ti_bq24190);
+
+static int bq24190_register_reset(const struct device *dev)
+{
+	const struct bq24190_config *const config = dev->config;
+	int ret, limit = 100;
```

Comment:
that 100 could probably use a define and then a comment referring to whatever number led to this one in the datasheet

Reply:
This was inherited from the Linux driver. I couldn't find anything in the datasheet to support the reset logic's timing characteristics, so this 100 * 100µs, so this seems to be more of a colloquial value.

My previous tenure was supporting this part, and this polling routine and interval seems appropriate.

I will give the 100 tries a define and throw in a comment.

Reply:
Oof, ok, this seems to be a pattern for this BQ parts, load of these in https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/bq274xx/bq274xx.c as well, same story, some timings just copied from the Linux driver, some @nixward going full archeology on the datasheets. Feel free to take inspiration from that one, had quite few eyes on it.